mujoco_warp/_src/solver_test.py

```diff
@@ -68,6 +69,69 @@ def cost(qacc):
 
       _assert_eq(mjwarp_cost, mj_cost, name="cost")
 
+  def test_init_linesearch(self):
+    """Test linesearch initialization."""
+    mjm, mjd, m, d = test_util.fixture(
+      "constraints.xml",
+      cone=ConeType.PYRAMIDAL,
+      iterations=0,
+      ls_iterations=0,
+    )
+
+    # One step to obtain more non-zeros results
+    mjwarp.step(m, d)
+
+    # Calculate target values
+    def calc_jv(nefc, efc_J, efc_search):
+      jv = np.zeros(nefc)
+      for i in range(nefc):
+        jv[i] += np.sum(efc_J[i, :] * efc_search[:])
+      return jv
+
+    def calc_quad_gauss(efc_gauss, efc_search, efc_Ma, qfrc_smooth, efc_mv):
+      quad_gauss = np.zeros(3)
+      quad_gauss[0] = efc_gauss[0]
+      quad_gauss[1] = np.sum(efc_search[:] * (efc_Ma[0, :] - qfrc_smooth[0, :]))
+      quad_gauss[2] = 0.5 * np.sum(efc_search[:] * efc_mv[:])
+      return quad_gauss
+
+    def calc_quad(nefc, efc_jaref, efc_jv, efc_D):
+      quad = np.zeros((nefc, 3))
```

Comment:
instead preallocating `quad` and then computing it by row, maybe something like:
```
quad = np.hstack([0.5 * efc_jaref * efc_jaref * efc_D, efc_jv * efc_jaref * efc_D, 0.5 * efc_jv * efc_jv * efc_D])
```